tests/integration/account_balance_query_e2e_test.py

```
@@ -1,14 +1,107 @@
+
+        assert balance is not None
+        assert hasattr(balance, "hbars")
+        assert balance.hbars.to_tinybars() >= 0
+    finally:
+        env.close()
+
+
+def test_integration_balance_query_raises_when_neither_source_set():
+    env = IntegrationTestEnv()
+    try:
+        with pytest.raises(ValueError, match="Either Account ID or Contract ID must be set before making the request."):
```

Ruff (0.14.10)

Pattern passed to `match=` contains metacharacters but is neither escaped nor raw

(RUF043)